can-list.js

```diff
@@ -797,7 +799,7 @@ assign(List.prototype, {
 		var item;
 		for (var i = 0, len = this.attr("length"); i < len; i++) {
 			item = this.attr(i);
-			if (cb.call(thisarg || item, item, i, this) === false) {
+			if (typeof item !== "undefined" && cb.call(thisarg || item, item, i, this) === false) {
```

Comment:
This could probably be `item !== undefined && ...`

What if there was a list like `new List([ true, undefined, undefined, true ])`.  What should the behavior be?